src/app/views/form/form.component.html

```diff
@@ -1,57 +1,88 @@
-    <label id="example-radio-group-label" class="required-after">Technical skills</label>
+
+    <label for="techSkills" id="example-radio-group-label" class="required-after">Technical skills</label>
     <p>Please evaluate the technical skills that the student has after the classes.</p>
+    <mat-error *ngIf="!isValid && techSkills.hasError('required')">
+      Please choose one of the options.
+    </mat-error>
     <mat-radio-group
+      id="techSkills"
       aria-labelledby="example-radio-group-label"
-      class="example-radio-group" formControlName="techSkills">
+      class="example-radio-group"
+      formControlName="techSkills"
+    >
       <mat-radio-button value="1">Beginner</mat-radio-button>
       <mat-radio-button value="2">Intermediate</mat-radio-button>
       <mat-radio-button value="3">Advanced</mat-radio-button>
     </mat-radio-group>
-    <label id="example-radio-group-label" class="required-after">Learning pace</label>
+
+    <label for="learningPace" id="example-radio-group-label" class="required-after">Learning pace</label>
     <p>Please evaluate the learning pace of the student.</p>
+    <mat-error *ngIf="!isValid && learningPace.hasError('required')">
+      Please choose one of the options.
+    </mat-error>
     <mat-radio-group
+      id="learningPace"
       aria-labelledby="example-radio-group-label"
-      class="example-radio-group" formControlName="learningPace">
+      class="example-radio-group"
+      formControlName="learningPace"
+    >
       <mat-radio-button value="1">Slow</mat-radio-button>
       <mat-radio-button value="2">Medium</mat-radio-button>
       <mat-radio-button value="3">Fast</mat-radio-button>
     </mat-radio-group>
-    <label id="example-radio-group-label" class="required-after">Extra mile</label>
+
+    <label for="extraMile" id="example-radio-group-label" class="required-after">Extra mile</label>
     <p>Please evaluate the degree to which the student was proactive, took more responsibilities than were required.</p>
+    <mat-error *ngIf="!isValid && extraMile.hasError('required')">
+      Please choose one of the options.
+    </mat-error>
     <mat-radio-group
+      id="extraMile"
       aria-labelledby="example-radio-group-label"
-      class="example-radio-group" formControlName="extraMile">
+      class="example-radio-group"
+      formControlName="extraMile"
+    >
       <mat-radio-button value="1">None</mat-radio-button>
       <mat-radio-button value="2">Some</mat-radio-button>
       <mat-radio-button value="3">A lot</mat-radio-button>
     </mat-radio-group>
+
+    <label id="example-radio-group-label" class="required-after">Comment</label>
+    <div>
+      {{ charsRemaining$ | async }}/{{ maxCharacters }} characters left.
+    </div>
     <mat-form-field appearance="fill">
       <mat-label>Comment</mat-label>
-      <textarea matInput formControlName="comment"></textarea>
+      <textarea matInput formControlName="comment" maxlength="150"></textarea>
```

Comment:
If you are using reactive forms, use it fully. You can move max length to form creation. Angular forms have `MaxLengthValidator`

Reply:
Updated with the solution in the documentation https://angular.io/api/forms/MaxLengthValidator